lib/segment/src/types.rs

```diff
@@ -1947,6 +1950,24 @@ pub enum Condition {
     Nested(NestedCondition),
     /// Nested filter
     Filter(Filter),
+
+    #[serde(skip)]
+    Resharding(Arc<dyn ReshardingCondition + Send + Sync + 'static>),
+}
+
+impl PartialEq for Condition {
+    fn eq(&self, other: &Self) -> bool {
+        match (self, other) {
+            (Self::Field(this), Self::Field(other)) => this == other,
+            (Self::IsEmpty(this), Self::IsEmpty(other)) => this == other,
+            (Self::IsNull(this), Self::IsNull(other)) => this == other,
+            (Self::HasId(this), Self::HasId(other)) => this == other,
+            (Self::Nested(this), Self::Nested(other)) => this == other,
+            (Self::Filter(this), Self::Filter(other)) => this == other,
+            (Self::Resharding(this), Self::Resharding(other)) => this.eq(other.deref()),
```

Comment:
Without `ReshardingCondition::eq` and `ReshardingCondition::as_any`, we can always return `false` here, but this will kinda break equality for `Condition`.